system/system_monitor/reader/hdd_reader/hdd_reader.cpp

```diff
@@ -220,14 +220,14 @@ int get_ata_identify(int fd, HDDInfo * info)
   // IDENTIFY DEVICE
   // Word 10..19 Serial number
   char serial_number[20 + 1]{};
-  strncpy(serial_number, reinterpret_cast<char *>(data) + 20, 20);
+  memcpy(serial_number, data + 20, 20);
```

Comment:
Is `serial_number` meant to be null-terminated? If so, it'd better to replace `strncpy` with `strlcpy` instead, as a safer replacement.

Reply:
@esteve On Ubuntu, `libbsd-dev` is required for `strlcpy`, but I could not find rosdep dependency definition in [ROS Index](https://index.ros.org/deps/). Do you know how to resolve it?

Reply:
@ito-san we'd need to submit a PR adding this entry to `rosdistro`. However, given that this is C++, we could use [string::copy](https://en.cppreference.com/w/cpp/string/basic_string/copy) or a similar method. What do you think?

Reply:
@esteve changed to use std::string constructor. How about this way?